content/en/blog/2023/contributing-to-otel-demo/index.md

```diff
@@ -0,0 +1,1026 @@
+  }),
+  resourceDetectors: [
+    containerDetector,
+    envDetector,
+    hostDetector,
+    osDetector,
+    processDetector,
+    alibabaCloudEcsDetector,
+    awsEksDetector,
+    awsEc2Detector,
+    gcpDetector,
+  ],
+});
+sdk.start();
+```
+
+> _View automatic instrumentation samples for other programming languages,
+> [here](/docs/instrumentation/)._
+
+These few lines of code will automatically start generating trace spans for your
+service.
+
+The
+[`paymentservice`](https://github.com/kubeshop/opentelemetry-demo/tree/main/src/paymentservice)
+also includes manually instrumented spans as well. If you take a look at the
+[`src/paymentservice/index.js`](https://github.com/kubeshop/opentelemetry-demo/blob/main/src/paymentservice/index.js)
+and
+[`src/paymentservice/charge.js`](https://github.com/kubeshop/opentelemetry-demo/blob/main/src/paymentservice/charge.js)
+you’ll see a require statement for the `@opentelemetry/api`, and a tracer that
+creates a span with some attributes.
+
+```jsx
+// src/paymentservice/index.js
+
+// Line 6
+const { trace, context, SpanStatusCode } = require('@opentelemetry/api');
+const tracer = trace.getTracer('paymentservice');
+
+// [...]
+
+function chargeServiceHandler(call, callback) {
+  // Line 23
+  const span = trace.getActiveSpan();
+  try {
+    const amount = call.request.amount;
+
+    span.setAttributes({
+      'app.payment.amount': parseFloat(`${amount.units}.${amount.nanos}`),
+    });
+    span.addEvent('Charge request received.', {
+      'log.severity': 'info',
+      'log.message': 'Charge request received.',
+      request: call.request,
+    });
+
+    const response = charge.charge(call.request);
+
+    // [...]
+  } catch (err) {
+    // [...]
+  }
+}
+
+// src/paymentservice/charge.js
+// Line 4
+const opentelemetry = require('@opentelemetry/api');
+const { context, propagation, trace, metrics } = opentelemetry;
+// [...]
+// Line 10
+const tracer = trace.getTracer('paymentservice');
+// [...]
+// Line 21
+const span = trace.getActiveSpan();
+// [...]
+// Line 53
+span.setAttributes({
+  'app.payment.card_type': cardType,
+  'app.payment.card_valid': valid,
+});
+// [...]
+```
+
+I’ll walk you through these three demos, with one sample without telemetry.
+
+1. **Demo 0**: What’s it like without OpenTelemetry traces?
+2. **Demo 1**: Get the active span from the context and use it as the main span
+   in the `chargeServiceHandler` .
+3. **Demo 2**: Get the active span from the context to create a new context.
+   Create a new span for the `chargeServiceHandler` and pass the new context in
+   as a parameter.
+4. **Demo 3**: Create a new active span for the `chargeServiceHandler` without
+   the need to pass a parent span and context.
+
+But first, let’s make it harder! It’s easy to get started when somebody is
+holding your hand. Let’s remove the guardrails for a second. Literally, let’s
+remove the OpenTelemetry instrumentation and run some API tests to see what
+happens.
```

Comment:
This belongs to the very top of the blog post, you should explain people what they are going to do. It's promised that they learn how to contribute, but the core of the blog post starts here.